lms/djangoapps/teams/views.py

```diff
@@ -0,0 +1,448 @@
+            del result_filter['is_active']
+        if 'text_search' in request.QUERY_PARAMS:
+            return Response({'detail': "text_search is not yet supported"}, status=status.HTTP_400_BAD_REQUEST)
+
+        queryset = CourseTeam.objects.filter(**result_filter)
+
+        order_by_input = request.QUERY_PARAMS.get('order_by', 'name')
+        if order_by_input == 'name':
+            queryset = queryset.extra(select={'lower_name': "lower(name)"})
+            order_by_field = 'lower_name'
+        elif order_by_input == 'open_slots':
+            queryset = queryset.annotate(team_size=Count('users'))
+            order_by_field = 'team_size'
+        elif order_by_input == 'last_activity':
+            return Response({'detail': "last_activity is not yet supported"}, status=status.HTTP_400_BAD_REQUEST)
+
+        queryset = queryset.order_by(order_by_field)
+
+        if not queryset:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        page = self.paginate_queryset(queryset)
+        serializer = self.get_pagination_serializer(page)
+        return Response(serializer.data)  # pylint: disable=maybe-no-member
+
+    def post(self, request):
+        """
+        POST /api/team/v0/teams/
+        """
+        field_errors = {}
+        course_key = None
+
+        try:
+            course_key = CourseKey.from_string(request.DATA.get('course_id'))
+            get_course(course_key)
+        except InvalidKeyError:
+            field_errors['course_id'] = {
+                'developer_message': "course_id is not valid.",
+                'user_message': _("course_id is not valid"),
+            }
+        except ValueError:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        if course_key and not (CourseEnrollment.is_enrolled(request.user, course_key) or request.user.is_staff):
+            return Response({'detail': "User must be enrolled."}, status=status.HTTP_403_FORBIDDEN)
+
+        data = request.DATA
+        data['course_id'] = course_key
+
+        serializer = CourseTeamCreationSerializer(data=data)
+        add_serializer_errors(serializer, data, field_errors)
+
+        if field_errors:
+            return Response({
+                'field_errors': field_errors,
+            }, status=status.HTTP_400_BAD_REQUEST)
+        else:
+            team = serializer.save()
+            return Response(CourseTeamSerializer(team).data)
+
+
+class TeamsDetailView(RetrievePatchAPIView):
+    """
+        **Use Cases**
+
+            Get or update a course team's information. Updates are supported
+            only through merge patch.
+
+        **Example Requests**:
+
+            GET /api/team/v0/teams/{team_id}}
+
+            PATCH /api/team/v0/teams/{team_id} "application/merge-patch+json"
+
+        **Response Values for GET**
+
+            If the user is logged in, the response contains the following fields:
```

Comment:
Yes, I think that makes sense, @benmcmorran. I'm now wondering whether we made a mistake by making teams have global ids, when what we are building is only course teams. If this is truly per course, then would things be simpler if we added that restriction? Is this different now that we're not using CourseUserGroup but building our own thing?